test/shared/utilities.ts

```diff
@@ -88,3 +92,19 @@ export async function mineBlock(provider: Web3Provider, timestamp: number): Prom
     );
   });
 }
+
+export async function mineBlocks(provider: Web3Provider, blockCount: number): Promise<void> {
+  for (let ind = 0; ind < blockCount; ind++) {
+    await mineBlock(provider);
+  }
+}
+
+export function convertStringToArrayish(data: string): Arrayish {
```

Comment:
Check, if solidity doesnt RLP encode by default without using this function.

Reply:
I previously checked in the contract IndexHybridToken.sol Asset.symbol is of type bytes8, without such conversion solidity gives an error:
Error: invalid input argument (arg="_assets", reason="invalid bytes8 value", value="BTC", version=4.0.47)

Maybe there is some kind of standard corvette function - but I didn't find it, so I added this function.

Reply:
OK. The thing is, your method has undefined behaviour when it needs to crash:
When you use `INVALID_ASSET`, you go out of bounds, since `Buffer.from('INVALID_ASSET') > 8`.
Let's use existing `ethers` functions instead of your implementation:
```Typescript
export function convertStringToArrayish(data: string): Arrayish {
  return ethers.utils.hexlify(ethers.utils.padZeros(ethers.utils.toUtf8Bytes(data), 8));
}
```
Then, we can rename `INVALID_ASSET` to `INVALID` to avoid out of bounds issues.

Reply:
In the **convertStringToArrayish** function, I did not check the size in bytes and allowed to make more than 8 bytes, I also specifically used INVALID_ASSET - for one test, where I showed that if more than 8 bytes, there will be an integrity error. Therefore, changing INVALID_ASSET to INVALID will not result in the desired error, so I will remove the test in which I check that the asset.symbol is greater than 8 bytes.

Using an existing **ethers** function is a good idea.